pkg/apis/integreatly/v1alpha1/grafana_types.go

```diff
@@ -13,21 +13,39 @@ import (
 type GrafanaSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	Hostname               string                  `json:"hostname,omitempty"`
-	Containers             []v1.Container          `json:"containers,omitempty"`
-	Secrets                []string                `json:"secrets,omitempty"`
-	DashboardLabelSelector []*metav1.LabelSelector `json:"dashboardLabelSelector,omitempty"`
-	LogLevel               string                  `json:"logLevel"`
-	AdminUser              string                  `json:"adminUser"`
 	AdminPassword          string                  `json:"adminPassword"`
-	BasicAuth              bool                    `json:"basicAuth"`
-	DisableLoginForm       bool                    `json:"disableLoginForm"`
-	DisableSignoutMenu     bool                    `json:"disableSignoutMenu"`
+	AdminUser              string                  `json:"adminUser"`
 	Anonymous              bool                    `json:"anonymous"`
+	BasicAuth              bool                    `json:"basicAuth"`
 	Config                 GrafanaConfig           `json:"config"`
+	Containers             []v1.Container          `json:"containers,omitempty"`
 	CreateRoute            bool                    `json:"createRoute"`
+	DashboardLabelSelector []*metav1.LabelSelector `json:"dashboardLabelSelector,omitempty"`
+	DisableLoginForm       bool                    `json:"disableLoginForm"`
+	DisableSignoutMenu     bool                    `json:"disableSignoutMenu"`
+	Hostname               string                  `json:"hostname,omitempty"`
+	Ingress                GrafanaIngress          `json:"ingress,omitempty"`
+	InitialReplicas        int                     `json:"initialReplicas,omitempty"`
+	LogLevel               string                  `json:"logLevel"`
+	Secrets                []string                `json:"secrets,omitempty"`
+	Service                GrafanaService          `json:"service,omitempty"`
+}
+
+// GrafanaService provides a means to configure the service
+type GrafanaService struct {
+	Annotations map[string]string    `json:"annotations,omitempty"`
+	Labels      metav1.LabelSelector `json:"labels,omitempty"`
```

Comment:
Why use a `LabelSelector` here instead of just `map[string]string`?

Reply:
make's perfect sense ... changing to a map